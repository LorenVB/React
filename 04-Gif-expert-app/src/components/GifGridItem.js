import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ( {title, url} ) => {
  
    return (
    <div>
       <img className="gif-img animate__animated animate__bounce animate__fadeIn" src={url} alt={ title } />
       <p> {title} </p>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}


