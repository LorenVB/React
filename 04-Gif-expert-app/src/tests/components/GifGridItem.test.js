import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Testing GifGridItem with Enzyme and Jest', () => {

    const title = "Testing Component";
    const url = "https:testing.com";

    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('should render the GifGridItem Component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should render the title prop as p html element', () => {
        const p = wrapper.find('p').text().trim();
        expect(p).toBe(title)
    })

    test('should render the alt and url tags with the url and title props', () => {
        const img = wrapper.find('img');
        console.log(img);
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    })

    test('should have animate__fadeIn class style', () => {
        const div = wrapper.find('div');
        console.log(div)
        const style = div.prop('className')
        expect(style.includes('animate__fadeIn')).toBe(true);
    })
})