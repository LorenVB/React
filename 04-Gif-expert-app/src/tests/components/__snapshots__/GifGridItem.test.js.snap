// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItem with Enzyme and Jest should render the GifGridItem Component 1`] = `
<div>
  <img
    alt="Testing Component"
    className="gif-img animate__animated animate__bounce animate__fadeIn"
    src="https:testing.com"
  />
  <p>
     
    Testing Component
     
  </p>
</div>
`;
